core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -348,7 +348,7 @@ class TeamPolicyInternal<Kokkos::Experimental::SYCL, Properties...>
   template <class FunctorType>
   int internal_team_size_recommended_reduce(const FunctorType& f) const {
     // FIXME_SYCL improve
-    return internal_team_size_max<ClosureType>(f);
+    return internal_team_size_max_reduce(f);
```

Comment:
OK that you will need to elaborate on.

Reply:
That's not changed here but in #3818. I rebased so that I could run the test you were commenting on. I'm happy to drop the commits coming from #3818 if you prefer to get this merged first.

Reply:
The reasoning here is that I really can't do all that much with the template argument and only really care if it's `parallel_for` or `parallel_reduce` (since we don't have parallel_scan for `TeamPolicy`).